pkg/entities/user_test.go

```diff
@@ -158,3 +158,144 @@ func TestEntity_SendGiftXp(t *testing.T) {
 		})
 	}
 }
+
+func TestEntity_GetUserProfile(t *testing.T) {
+	type fields struct {
+		repo     *repo.Repo
+		store    repo.Store
+		log      logger.Logger
+		dcwallet discordwallet.IDiscordWallet
+		discord  *discordgo.Session
+		cache    cache.Cache
+		svc      *service.Service
+		cfg      config.Config
+	}
+	type args struct {
+		guildID string
+		userID  string
+	}
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	cfg := config.Config{
+		DBUser: "postgres",
+		DBPass: "postgres",
+		DBHost: "localhost",
+		DBPort: "5434",
+		DBName: "mochi_local",
+
+		InDiscordWalletMnemonic: "holiday frequent toy bachelor auto use style result recycle crumble glue blouse",
+		FantomRPC:               "https://rpc.ftm.tools",
+		FantomScan:              "https://api.ftmscan.com/api?",
+		FantomScanAPIKey:        "XEKSVDF5VWQDY5VY6ZNT6AK9QPQRH483EF",
+
+		EthereumRPC:        "https://mainnet.infura.io/v3/5b389eb75c514cf6b1711d70084b0114",
+		EthereumScan:       "https://api.etherscan.io/api?",
+		EthereumScanAPIKey: "SM5BHYSNIRZ1HEWJ1JPHVTMJS95HRA6DQF",
+
+		BscRPC:        "https://bsc-dataseed.binance.org",
+		BscScan:       "https://api.bscscan.com/api?",
+		BscScanAPIKey: "VTKF4RG4HP6WXQ5QTAJ8MHDDIUFYD6VZHC",
+		DiscordToken:  "OTcxNjMyNDMzMjk0MzQ4Mjg5.G5BEgF.rv-16ZuTzzqOv2W76OljymFxxnNpjVjCnOkn98",
+
+		RedisURL: "redis://localhost:6379/0",
+	}
+
+	s := pg.NewPostgresStore(&cfg)
+	r := pg.NewRepo(s.DB())
+	uXp := mock_guild_user_xp.NewMockStore(ctrl)
+
+	userXP := model.GuildUserXP{
+		GuildID: "981128899280908299",
+		UserID:  "963641551416881183",
+	}
+
+	uXp.EXPECT().GetOne("981128899280908299", "963641551416881183").Return(&userXP, nil).AnyTimes()
+
+	uXp.EXPECT().GetOne("abc", "abc").Return(nil, errors.New("cannot find user")).AnyTimes()
+	uXp.EXPECT().GetOne("abc", "963641551416881183").Return(nil, errors.New("cannot find guild")).AnyTimes()
+
```

Comment:
missing mocks for `ConfigXPLevel` repo
![image](https://user-images.githubusercontent.com/12120998/172305072-9f733869-40c4-4d8e-8656-0da5932063fb.png)